src/backend/distributed/planner/multi_physical_planner.c

```diff
@@ -5717,7 +5717,7 @@ AssignDualHashTaskList(List *taskList)
 	 * if subsequent jobs have a small number of tasks, we won't allocate the
 	 * tasks to the same worker repeatedly.
 	 */
-	List *workerNodeList = ActiveReadableNonCoordinatorNodeList();
+	List *workerNodeList = ActiveReadableNodeList();
```

Comment:
We have other types  of repartition joins as well. We should at least make sure that single hash  repartition joins work fine. In that case, you need to do two things: (a) join should include table_1.dist_key = table_2.non_dist_key  (b) `SET citus.enable_single_hash_repartition_joins TO ON;`